src/lib/minions/functions/degradeableItemsCommand.ts

```diff
@@ -17,54 +17,80 @@ export async function degradeableItemsCommand(
 	const item = degradeableItems.find(i => [i.item.name, ...i.aliases].some(n => stringMatches(n, input ?? '')));
 	const number = mahojiParseNumber({ input: quantity, min: 1, max: 1_000_000 });
 
+	// If not input on `/minion charge` show the user what can be charged, the amounts, and their current charges
 	if (!input || !number || !item || number < 1 || number > 100_000) {
 		return `Use \`/minion charge item: [${degradeableItems.map(i => i.item.name).join('|')}] amount:[1-100,000]\`
-${degradeableItems
-	.map(i => {
-		const charges = user.user[i.settingsKey];
-		return `${i.item.name}: ${charges.toLocaleString()} charges`;
-	})
-	.join('\n')}`;
+    ${degradeableItems
+		.map(i => {
+			const charges = user.user[i.settingsKey];
+			return `${i.item.name}: ${charges.toLocaleString()} charges`;
+		})
+		.join('\n')}`;
 	}
 
+	// Get the cost and amount of charges
 	const cost = item.chargeInput.cost.clone().multiply(number);
 	const amountOfCharges = item.chargeInput.charges * number;
 
+	// Check if the user has materials to charge the item
 	if (!user.owns(cost)) {
 		return `You don't own ${cost}.`;
 	}
 
+	// Check if the item needs converted and has a uncharged version
 	const needConvert = item.convertOnCharge && item.unchargedItem;
-	if (needConvert && !user.hasEquippedOrInBank(item.item.id) && !user.owns(item.unchargedItem!.id)) {
-		return `You don't own a ${item.item.name} or ${item.unchargedItem!.name}.`;
+
+	// Check for variants in the users bank
+	let unchargedItem = item.unchargedItem!;
+	let chargedItem = item.item;
+	for (const variant of item.itemVariants!) {
+		if (user.owns(variant.unchargedVariant.id)) {
+			unchargedItem = variant.unchargedVariant;
+			chargedItem = variant.chargedVariant;
+			break;
+		}
+	}
+
+	// Show error message if the user doesn't have the charged, uncharged, or variants of the item in the bank
+	if (!user.hasEquippedOrInBank(chargedItem.id) && !user.owns(unchargedItem.id)) {
+		return `You don't own a ${chargedItem.name}${
+			unchargedItem!.name === chargedItem.name ? '' : `, ${unchargedItem!.name}`
```

Comment:
I think this conditional could have been handled better, if the uncharged and charged are the same name that property can be removed due to it not being required.

https://github.com/oldschoolgg/oldschoolbot/blob/5f7e55b97acace2635919ae3bb3b076df44f8919/src/lib/degradeableItems.ts#L12-L39C2

This would mean modifying how the break code works so that it gives back the item if no uncharged version exists

Reply:
Yeah I was trying to make an all in one error for if the user didn't have an item, like ash sanctifier. I just made a new error for that case since this seems to break too many things and lacks readability.

Reply:
This is also what caused the blood essence issue, and has been fixed.